ceph/rados/core_workflows.py

```diff
@@ -1498,6 +1498,31 @@ def fetch_host_node(self, daemon_type: str, daemon_id: str = None) -> object:
             )
             return None
 
+    def fetch_host_config(self, daemon_type: str, daemon_id: str = None) -> object:
+        """
+        Fetches the config of the given daemon in a cluster
+
+        Args:
+            daemon_type (str): type of daemon
+            daemon_id (str, optional): name of the daemon, ID in case of OSD's
+
+        Returns:
+            dict: host config
+        """
+        json_str = None
+        if daemon_type == "osd":
+            cmd = f"cephadm shell -- ceph daemon {daemon_type}.{daemon_id} config show --format json"
+            osd_node = self.fetch_host_node(daemon_type, daemon_id)
+            json_str = osd_node.exec_command(cmd)
+        else:
+            cmd = (
+                f"cephadm shell -- ceph daemon {daemon_type} config show --format json"
+            )
+            json_str = self.run_ceph_command(cmd)
```

Comment:
`run_ceph_command` method will only run on installer node. The command needs to be run on host where the daemon is present.